File: Problem Set 1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)

{
    float dollars;
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    do
    {
        //Prompting the user to enter the amount of change owed
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);
    //Converting dollars to cents to avoid imprecision
    int cents = round(dollars * 100);
    int coins = 0;
    //Using quarters if possible
    while (cents >= quarter)
    {
        cents = cents - quarter;
        coins++;
    }
    //Using dimes if possible
    while (cents >= dime && cents < quarter)
    {
        cents = cents - dime;
        coins++;
    }
    //Using nickels if possible
    while (cents >= nickel && cents < dime)
    {
        cents = cents - nickel;
        coins++;
    }
    //Using pennies if possible
    while (cents >= penny && cents < nickel)
    {
        cents = cents - penny;
        coins++;
    }
    //Showing the amount of coins
    printf("%i\n", coins);
}

File: Problem Set 6/cash.py
from cs50 import get_float

# Define the coins
quarter = 25
dime = 10
nickel = 5
penny = 1

while True:
    # Prompt user for input
    change = get_float("Change owed: ")
    if change >= 0:
        change *= 100
        count = 0
        # Calculate quantity for each coin
        while change > 0:
            if change >= quarter:
                count += int(change / quarter)
                change = change % quarter
            if change >= dime:
                count += int(change / dime)
                change = change % dime
            if change >= nickel:
                count += int(change / nickel)
                change = change % nickel
            if change >= penny:
                count += int(change / penny)
                change = change % penny
        print(count)
        break